usaspending_api/etl/management/commands/combine_transaction_search_chunks.py

```diff
@@ -0,0 +1,218 @@
+
+        if not options["keep_old_data"]:
+            with Timer("Clearing old table"):
+                self.remove_old_data(chunk_count)
+
+        if not options["keep_matview_data"]:
+            with Timer("Emptying Matviews"):
+                self.empty_matviews()
+
+        with Timer("Granting Table Permissions"):
+            self.grant_matview_permissions()
+
+    def read_constraint_definitions(self):
+        indexes_sql = (self.matview_dir / "componentized" / "transaction_search__constraints.sql").read_text()
+
+        with connection.cursor() as cursor:
+            cursor.execute(indexes_sql)
+            rows = cursor.fetchall()
+
+        create_temp_constraints = []
+        rename_constraints_old = []
+        rename_constraints_temp = []
+        for row in rows:
+            constraint_name = row[0]
+            constraint_name_temp = constraint_name + "_temp"
+            constraint_name_old = constraint_name + "_old"
+
+            create_temp_constraints.append(
+                f"ALTER TABLE public.{TABLE_NAME}_temp ADD CONSTRAINT {constraint_name_temp} {row[1]};"
+            )
+            rename_constraints_old.append(
+                f"ALTER TABLE public.{TABLE_NAME}_old RENAME CONSTRAINT {constraint_name} TO {constraint_name_old};"
+            )
+            rename_constraints_temp.append(
+                f"ALTER TABLE public.{TABLE_NAME} RENAME CONSTRAINT {constraint_name_temp} TO {constraint_name};"
+            )
+
+            self.constraint_names.append(constraint_name)
+
+        return create_temp_constraints, rename_constraints_old + rename_constraints_temp
+
+    def read_index_definitions(self):
+        indexes_sql = (self.matview_dir / "componentized" / "transaction_search__indexes.sql").read_text()
+
+        with connection.cursor() as cursor:
+            cursor.execute(indexes_sql)
+            rows = cursor.fetchall()
+
+        create_temp_indexes = []
+        rename_indexes_old = []
+        rename_indexes_temp = []
+        for row in rows:
+            index_name = row[0]
+            index_name_temp = index_name + "_temp"
+            index_name_old = index_name + "_old"
+
+            # Ensure that the index hasn't already been created by a constraint
+            if index_name not in self.constraint_names:
+                create_temp_index_sql = row[1].replace(index_name, index_name_temp)
+                create_temp_index_sql = create_temp_index_sql.replace(
+                    f"public.{TABLE_NAME}", f"public.{TABLE_NAME}_temp"
+                )
+                create_temp_indexes.append(create_temp_index_sql)
+
+                rename_indexes_old.append(f"ALTER INDEX IF EXISTS {index_name} RENAME TO {index_name_old};")
+                rename_indexes_temp.append(f"ALTER INDEX IF EXISTS {index_name_temp} RENAME TO {index_name};")
+
+        return create_temp_indexes, rename_indexes_old + rename_indexes_temp
+
+    def recreate_matview(self):
```

Comment:
A few function names could be altered for accuracy. Not a blocker for PR merge

Reply:
Updated:
- `recreate_matview()` -> `recreate_table()`
- `swap_matviews()` -> `swap_tables()`
- `grant_matview_permissions()` -> `grant_table_permissions`